### DISCUSSION TITLE
poetry.lock file hashes

### DISCUSSION THREAD

**Discussion by nayarudolph** [python-poetry/poetry#8026] [Q&A] [answered]
I am working with poetry for the first time and trying to add a package to a forked repo. 
When I do "poetry install" for the first time everything works out and the poetry.lock file is unchanged.
Now I wanted to add one and did "poetry add [package]" which resolves and installs the dependencies but it adds file names and hashes to the poetry.lock. 
Is that necesseray and can I disable this? The repo I forked has no hashes in their poetry.lock and it seems weird to only add them for the new packages.

Thank you for the help

**Comment by radoering** [comment, marked as answer]
> Is that necesseray and can I disable this?

It is necessary and that is why you cannot disable it. Poetry only installs dependencies that are included in the lockfile.

> The repo I forked has no hashes in their poetry.lock

Might be an invalid (or very old?) lockfile.

**Reply by nayarudolph** [reply]
Thank you for the quick answer. Then yes I guess no one has updated or added packages to this lockfile/repo for quite some while.
